hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceInterface.java

```diff
@@ -52,15 +52,17 @@
    * @param queueStorage the replication queue storage
    * @param replicationPeer the replication peer
    * @param server the server which start and run this replication source
+   * @param producer the name of region server which produce the replication queue
```

Comment:
This is for the RecoveryReplicationSource only?

Reply:
For all ReplicationSource. The assumption here is that: all replication is a producer-consumer module.